<?xml version="1.0"?>
<launch>
	<!-- VIDEO SOURCE -->
	<arg name="input" default="csi://0"/>
	<arg name="input_width" default="640"/>
	<arg name="input_height" default="480"/>
	<arg name="input_framerate" default="60.0"/>
	<arg name="input_codec" default="unknown"/>
	<arg name="input_loop" default="0"/>
	<arg name="input_latency" default="2000"/>
	<arg name="input_flip" default="horizontal-flip"/>
	
	<node pkg="ros_deep_learning" exec="video_source" output="screen">
		<param name="resource" value="$(var input)"/>
		<param name="width" value="$(var input_width)"/>
		<param name="height" value="$(var input_height)"/>
		<param name="codec" value="$(var input_codec)"/>
		<param name="loop" value="$(var input_loop)"/>
		<param name="latency" value="$(var input_latency)"/>
		<param name="framerate" value="$(var input_framerate)"/>
		<param name="flip" value="$(var input_flip)"/>
	</node>

    <!-- Detectnet -->
    <!-- <include file="$(find-pkg-share rrbot_cam_object_detector)/launch/detect_net.launch"/> -->
    
	<!-- HSV DETECTOR -->
    <include file="$(find-pkg-share rrbot_cam_object_detector)/launch/detect_hsv.launch.py"/>

	<!-- VIDEO OUTPUT -->
	<!-- <arg name="topic" value="/video_source/raw"/> -->
	<!-- <arg name="topic" value="/debug_image"/> -->
	<!-- <arg name="topic" value="/detectnet/overlay"/> -->
	<include file="$(find-pkg-share ros_deep_learning)/launch/video_output.ros2.launch">
		<arg name="topic" value="/debug_image"/>
        <arg name="output" value="webrtc://@:8554/output"/>
	</include>

</launch>